nexus/server.go

```diff
@@ -311,8 +326,21 @@ func (h *httpHandler) getOperationResult(writer http.ResponseWriter, request *ht
 	}
 	options := GetOperationResultOptions{Header: httpHeaderToNexusHeader(request.Header)}
 
-	waitStr := request.URL.Query().Get(queryWait)
+	// If both Request-Timeout http header and wait query string are set, the minimum will be used as the context timeout.
```

Comment:
This should be the minimum between `h.options.GetResultTimeout` and the parsed header value.